src/Forms/Form.php

```diff
@@ -1552,10 +1552,13 @@ public function forTemplate()
             HTTPCacheControlMiddleware::singleton()->disableCache();
         }
 
-        $return = $this->renderWith($this->getTemplates());
+        $context = $this;
+        $this->extend('onBeforeRender', $context);
+
+        $return = $context->renderWith($this->getTemplates());
```

Comment:
Would this also make sense then? Is there a possibility the form field could be overwritten in a way that could also provide a different set of templates?

```suggestion
        $return = $context->renderWith($context->getTemplates());
```

Reply:
Yeah I suppose so, I’ll push a commit that changes this for `Form` and `FormField` so it’s consistent